canopeum_backend/canopeum_backend/views.py

```diff
@@ -728,14 +728,12 @@ class AnnouncementDetailAPIView(APIView):
         operation_id="announcement_update",
     )
     def patch(self, request: Request, siteId):
-        try:
-            announcement = Announcement.objects.get(site=siteId)
-        except Announcement.DoesNotExist:
-            return Response(status=status.HTTP_404_NOT_FOUND)
+        announcement = Announcement.objects.get_or_create(site=siteId)
```

Comment:
Can't comment across deleted lines, but here I think it'd make more sense to index once (this goes with the below comment)
```suggestion
        announcement = Announcement.objects.get_or_create(site=siteId)[0]
```